includes/class-omise-callback.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 4.0
+ */
+class Omise_Callback {
+	/**
+	 * @var \WC_Abstract_Order
+	 */
+	protected $order;
+
+	/**
+	 * @var \OmiseCharge
+	 */
+	protected $charge;
+
+	/**
+	 * @param \WC_Abstract_Order $order
+	 */
+	public function __construct( $order ) {
+		$this->order = $order;
+		if ( ! $this->order || ! $this->order instanceof WC_Abstract_Order ) $this->invalid_result();
+	}
+
+	public static function execute() {
+		$order_id = isset( $_GET['order_id'] ) ? sanitize_text_field( $_GET['order_id'] ) : null;
+
+		$callback = new self( wc_get_order( $order_id ) );
+		$callback->validate();
+	}
+
+	public function validate() {
```

Comment:
Not confident if this is a perfect name. Naming ideas are welcomed :D

Reply:
Seems fine to me